### ISSUE TITLE
Save optimizer trace

### ISSUE THREAD

**Issue by paulstapor** [ICB-DCM/pyPESTO#108]
Currently, no optimizer trace is/can be saved. This could be done via the callback-input in the minimize-framework...

**Comment by yannikschaelte** [comment]
Actually, the trace can be saved already. The code for this is in https://github.com/ICB-DCM/pyPESTO/blob/master/pypesto/objective/history.py. Activating it works via the Objective options.

We had decided to implement an own trace and not use e.g. the Scipy ones, in order to have the same quality of information independent of the used optimizer. This is because the optimizers decide in many different ways e.g. what they consider a function evaluation, or when to append to the trace, so we did not want to use this, but rather decide ourselves what belongs in the trace.
